File: WebApi/Models/Entities/ExtendedRepo.cs
namespace WebApi.Models.Models
{
	/// <summary>wrapper over repo, with app settings like favorability.<para />
	/// note: composition over inheritance</summary>
	public class ExtendedRepo
	{
		/// <summary>repo</summary>
		public Repo Repo { get; set; } = new Repo();

		/// <summary>is favorite (for the current user)?</summary>
		public bool IsFavorite { get; set; } = false;
	}
}

File: ClientPages/src/app/models/extended-repo.ts
import { Repo } from "./repo.model";

export class ExtendedRepo {
	/** unique identifier (owner/name, i.e. facebook/react) */
	public get fullName(): string {
		return this.repo.fullName;
	}

	/** repo */
	public repo: Repo = new Repo();

	/** is favorite (for the current user)? */
	public isFavorite: boolean = false;
}
